File: linux/ardium_source/presentation/06_FileIO/file_io.m
// Topic: File I/O
// Write to test.txt

#import <Foundation/Foundation.h>

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        NSString *str = @"Hello Obj-C File IO";
        NSString *filename = @"test.txt";
        NSError *error;
        
        [str writeToFile:filename atomically:YES encoding:NSUTF8StringEncoding error:&error];
        
        if (error) {
            NSLog(@"Failed to open file");
        } else {
            NSLog(@"Written to test.txt");
        }
    }
    return 0;
}

File: linux/ardium_source/presentation/06_FileIO/file_io.go
// Topic: File I/O
// Write to test.txt

package main

import (
	"fmt"
	"os"
)

func main() {
	d1 := []byte("Hello Go File IO")
	err := os.WriteFile("test.txt", d1, 0644)
	if err != nil {
		fmt.Println("Failed to open file")
		return
	}
	fmt.Println("Written to test.txt")
}
